contracts/dao/ThenaStaking.sol

```diff
@@ -187,4 +206,24 @@ contract ThenaStaking is OwnableUpgradeable, ReentrancyGuardUpgradeable {
     function setHarvestTimeGap(uint256 _harvestTimeGap) external onlyOwner {
         harvestTimeGap = _harvestTimeGap;
     }
+
+    /**
+      * @dev emergency withdraw all lp tokens from farming contract given the list of lp token addresses
+      * @param lpTokens the list of lp token addresses
+      */
+    function emergencyWithdraw(address[] memory lpTokens) external onlyOwner nonReentrant {
+        for (uint256 i = 0; i < lpTokens.length; ++i) {
+            address lpToken = lpTokens[i];
+            Pool memory poolInfo = pools[lpToken];
+            if (!poolInfo.isActive) {
+                continue; // Skip inactive pool
+            }
+            uint256 lpAmount = IERC20(poolInfo.lpToken).balanceOf(address(this));
+            IGaugeV2(poolInfo.poolAddress).emergencyWithdraw();
+            lpAmount = IERC20(poolInfo.lpToken).balanceOf(address(this)) - lpAmount;
+
+            emit EmergencyWithdraw(poolInfo.poolAddress, lpAmount);
+        }
+        emergencyMode = true;
```

Comment:
add event

Reply:
done in 49ec5670cb578e687c2bd3d9cd4fabf31f7d2863